In [2]:
import pandas as pd

file_path = 'food_nutrient.csv'

# Read only the first 1000 rows to inspect data types
sample_df = pd.read_csv(file_path, nrows=1000)

# See the memory usage and default data types
print(sample_df.info(memory_usage='deep'))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 13 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   id                   1000 non-null   int64  
 1   fdc_id               1000 non-null   int64  
 2   nutrient_id          1000 non-null   int64  
 3   amount               1000 non-null   float64
 4   data_points          0 non-null      float64
 5   derivation_id        1000 non-null   int64  
 6   min                  0 non-null      float64
 7   max                  0 non-null      float64
 8   median               0 non-null      float64
 9   loq                  0 non-null      float64
 10  footnote             0 non-null      float64
 11  min_year_acquired    0 non-null      float64
 12  percent_daily_value  716 non-null    float64
dtypes: float64(9), int64(4)
memory usage: 101.7 KB
None


In [3]:
sample_df

id   fdc_id  nutrient_id  amount  data_points  derivation_id  min  \
0    13706927  1105904         1257    0.00          NaN             71  NaN   
1    13706930  1105904         1293   53.33          NaN             71  NaN   
2    13706926  1105904         1253    0.00          NaN             75  NaN   
3    13706921  1105904         1092    0.00          NaN             75  NaN   
4    13706916  1105904         1008  867.00          NaN             71  NaN   
..        ...      ...          ...     ...          ...            ...  ...   
995  13708130  1105968         1004    0.00          NaN             75  NaN   
996  13708129  1105968         1003    0.00          NaN             71  NaN   
997  13708131  1105968         1005   11.18          NaN             71  NaN   
998  13708139  1105968         1110    0.00          NaN             75  NaN   
999  13708133  1105968         2000   11.18          NaN             71  NaN   

     max  median  loq  footnote  min_year_acquired  percent_daily_value  
0    NaN     NaN  NaN       NaN                NaN                  NaN  
1    NaN     NaN  NaN       NaN                NaN                  0.0  
2    NaN     NaN  NaN       NaN                NaN                  0.0  
3    NaN     NaN  NaN       NaN                NaN                  0.0  
4    NaN     NaN  NaN       NaN                NaN                  NaN  
..   ...     ...  ...       ...                ...                  ...  
995  NaN     NaN  NaN       NaN                NaN                  0.0  
996  NaN     NaN  NaN       NaN                NaN                  NaN  
997  NaN     NaN  NaN       NaN                NaN                 14.0  
998  NaN     NaN  NaN       NaN                NaN                  0.0  
999  NaN     NaN  NaN       NaN                NaN                  NaN  

[1000 rows x 13 columns]

In [9]:
print(sample_df["nutrient_id"][0])
print(sample_df["nutrient_id"][1])

1257
1293


In [7]:
file_path = 'nutrient.csv'

# Read only the first 1000 rows to inspect data types
nutrient = pd.read_csv(file_path, nrows=1000)

print(nutrient.info(memory_usage='deep'))

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 477 entries, 0 to 476
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            477 non-null    int64  
 1   name          477 non-null    object 
 2   unit_name     477 non-null    object 
 3   nutrient_nbr  465 non-null    float64
 4   rank          466 non-null    float64
dtypes: float64(2), int64(1), object(2)
memory usage: 64.6 KB
None


In [8]:
nutrient

id                                        name unit_name  nutrient_nbr  \
0    2047            Energy (Atwater General Factors)      KCAL         957.0   
1    2048           Energy (Atwater Specific Factors)      KCAL         958.0   
2    1001                                      Solids         G         201.0   
3    1002                                    Nitrogen         G         202.0   
4    1003                                     Protein         G         203.0   
..    ...                                         ...       ...           ...   
472  2064                            Oligosaccharides        MG           NaN   
473  2065  Low Molecular Weight Dietary Fiber (LMWDF)         G         293.4   
474  2068                                   Vitamin E        MG         959.0   
475  2067                                   Vitamin A        UG         960.0   
476  2069                                 Glutathione        MG         961.0   

       rank  
0     280.0  
1     290.0  
2     200.0  
3     500.0  
4     600.0  
..      ...  
472  2250.0  
473  1306.0  
474  7810.0  
475  7430.0  
476  9000.0  

[477 rows x 5 columns]

In [29]:
nutrient['id'][0]

2047

In [32]:
for i in range(len(nutrient['id'])):
    if nutrient['id'][i] == 1257 or nutrient['id'][i] == 1293:
        print(i)

258
293


In [34]:
nutrient.loc[258]

id                                  1257
name            Fatty acids, total trans
unit_name                              G
nutrient_nbr                       605.0
rank                             15400.0
Name: 258, dtype: object

In [35]:
nutrient.loc[293]

id                                            1293
name            Fatty acids, total polyunsaturated
unit_name                                        G
nutrient_nbr                                 646.0
rank                                       12900.0
Name: 293, dtype: object

In [11]:
file_path2 = 'food.csv'

# Read only the first 1000 rows to inspect data types
food_names = pd.read_csv(file_path2, nrows=1000)

food_names

fdc_id          data_type  \
0    1105904       branded_food   
1    1105905       branded_food   
2    1105906       branded_food   
3    1105907       branded_food   
4    1105898  experimental_food   
..       ...                ...   
995  1106901       branded_food   
996  1106902       branded_food   
997  1106903       branded_food   
998  1106904       branded_food   
999  1106905       branded_food   

                                           description       food_category_id  \
0                           WESSON Vegetable Oil 1 GAL            Oils Edible   
1                                   SWANSON BROTH BEEF  Herbs/Spices/Extracts   
2             CAMPBELL'S SLOW KETTLE SOUP CLAM CHOWDER         Prepared Soups   
3          CAMPBELL'S SLOW KETTLE SOUP CHEESE BROCCOLI         Prepared Soups   
4    Discrepancy between the Atwater factor predict...                    NaN   
..                                                 ...                    ...   
995            ENRICHED MACARONI PRODUCT, SMALL SHELLS  Pasta by Shape & Type   
996                   ENRICHED MACARONI PRODUCT, STARS  Pasta by Shape & Type   
997                 ENRICHED MACARONI PRODUCT, GEMELLI  Pasta by Shape & Type   
998                ENRICHED MACARONI PRODUCT, LINGUINE  Pasta by Shape & Type   
999          ENRICHED MACARONI PRODUCT, ROMAN RIGATONI  Pasta by Shape & Type   

    publication_date  
0         2020-11-13  
1         2020-11-13  
2         2020-11-13  
3         2020-11-13  
4         2020-10-30  
..               ...  
995       2020-11-13  
996       2020-11-13  
997       2020-11-13  
998       2020-11-13  
999       2020-11-13  

[1000 rows x 5 columns]

The "food_nutrient.csv" file contains the ids of the most of the foods within the FoodCentral database. For each food, there will be multiple entries(rows) for a single food (fdc_id), providing the "nutrient_id". This "nutrient_id" can then be related to its values for macro/micro-nutrients contained in the 'nutrient.csv'


for example: sample_df["fdc_id"][0] and sample_df["fdc_id"][1] provide an fdc_id for 1105904 -> these are entries for the same food, but for different macro-nutrients


import pandas as pd

file_path = 'food_nutrient.csv'

sample_df = pd.read_csv(file_path, nrows=1000)

sample_df["nutrient_id"][0] is equivalent to 1257	 
sample_df["nutrient_id"][1] is equivalent to 1293

We take this "nutrient_id" and retrieve the related information in the nutrient.csv file

file_path = 'nutrient.csv'
nutrient.loc[258] -> Fatty acids, total trans in GRAMS
nutrient.loc[293] -> Fatty acids, total polyunsaturated in GRAMS


With this understanding, take these steps to make the python script:
1. Make a new csv file called Nutrition_Database
2. In the Nutrition_Database.csv, create new unique features for the values found in the 'food_nutrient.csv' column "nutrient_id", naming the header the with that unique nutrient_id value. Similarly, create unique rows for unique "fdc_id" values.
3. For each feature generated, replace the nutrient_id header with its corresponding entry in the "name" column listed in the nutrient.csv file (in the nutrient.csv file, column containing nutrient_id values is named "id"). Additionally, to the end of the renamed feature, add "_unit_name", getting the "unit_named" from the corresponding entry in the "unit_name" column in the nutrient.csv file.
4. Similarly, for all the fdc_id entries in the Nutrition_Database.csv, replace the fdc_id values with the corresponding value found in the "food.csv" in the "description" column (there is a column in the food.csv named "fdc_id"). Additionally, add the features "food_category_id" and the "publication_date" to their corresponding entries.

Accomplish all this by creating a sci-kit learn pipeline, making function defintions for this process.
